pandas/_libs/groupby.pyx

```diff
@@ -1414,25 +1435,32 @@ cdef masked_cummin_max(groupby_t[:, ::1] out,
     cdef:
         Py_ssize_t i, j, N, K
         groupby_t val, mval
+        uint8_t[::1] seen_na
         intp_t lab
 
     N, K = (<object>values).shape
+    seen_na = np.zeros((<object>accum).shape[0], dtype=np.uint8)
```

Comment:
can we avoid initializing this in some cases?

Reply:
Not sure here since we can't tell if a missing value is present without looking through the whole array. I suppose we could only allocate as soon as a missing value is seen, but that would add some complexity since algo is wrapped in a `nogil`.

Hopefully this is not a big memory addition since it is `ngroups * ncols` bytes vs the `nrow * ncols * 4` for the values (since `groupby_t` only covers (int64/float64/32)